charts/seed-bootstrap/charts/loki/templates/hvpa.yaml

```
@@ -0,0 +1,82 @@
+{{ if .Values.hvpa.enabled }}
+{{- /* .Values.replicas is of type 'float64', so let's cast it to string to have proper types for comparison */}}
+{{- if ne (.Values.replicas | toString) "0" }}
+apiVersion: autoscaling.k8s.io/v1alpha1
+kind: Hvpa
+metadata:
+  name: loki
+  namespace: {{ .Release.Namespace }}
+  labels:
+{{ toYaml .Values.labels | indent 4 }}
+spec:
+  replicas: 1
+{{- if .Values.hvpa.maintenanceWindow }}
+  maintenanceTimeWindow:
+{{ toYaml .Values.hvpa.maintenanceWindow | indent 4 }}
+{{- end }}
+  hpa:
+    selector:
+      matchLabels:
+        role: loki-hpa
+    deploy: false
+    template:
+      metadata:
+        labels:
+          role: loki-hpa
+      spec:
+        maxReplicas: {{ .Values.replicas }}
+        minReplicas: {{ .Values.replicas }}
+        metrics:
+        - resource:
+            name: memory
+            targetAverageUtilization: {{ .Values.hvpa.targetAverageUtilizationMemory }}
+          type: Resource
+        - resource:
+            name: cpu
+            targetAverageUtilization: {{ .Values.hvpa.targetAverageUtilizationCpu }}
+          type: Resource
+  vpa:
+    selector:
+      matchLabels:
+        role: loki-vpa
+    deploy: true
+    scaleUp:
+      updatePolicy:
+        updateMode: "Auto"
+{{- if .Values.hvpa.scaleUpStabilization }}
+{{ toYaml .Values.hvpa.scaleUpStabilization | indent 6 }}
+{{- end }}
+    scaleDown:
+      updatePolicy:
+        updateMode: "MaintenanceWindow"
```

```suggestion
        updateMode: "Auto"
```
We decided to scaled down ETCD only during maintenance window because of the disruption to the control-plane. It might be keep automatic scale down for loki. WDYT?